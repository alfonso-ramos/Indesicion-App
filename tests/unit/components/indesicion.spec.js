import { shallowMount } from "@vue/test-utils";
import Indecision from "@/components/Indecision.vue";

describe('Indesicion component', () => {
    let wrapper;
    let clgSpy;
    beforeEach(() => {
        wrapper = shallowMount(Indecision)

        clgSpy = jest.spyOn(console, 'log')
    })
    test('Debe hacer match con el snapshot ', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });
    test('Al escribir en el input no debe de disparar nada (console.log)', async () => {
        const input = wrapper.find("input")
        await input.setValue("Hola mundo")

        expect(clgSpy).toHaveBeenCalled()
    });
    test('Al escribir "?" debe de disparar el fetch', () => {
        
    });
    test('Pruebas en getAnswer', () => {
        
    });
    test('Pruebas en getAnswer - fallos en la API', () => {
        
    });
})