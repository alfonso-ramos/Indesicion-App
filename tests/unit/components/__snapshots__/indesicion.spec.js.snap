// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indesicion component Debe hacer match con el snapshot  1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p class="reminder">Recuerda terminar con un signo de interrogacion (?)</p>
  <!--v-if-->
</div>
`;
